assets/js/src/core/components/modal/simple-modal/hooks/use-simple-modal.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+* Pimcore
+*
+* This source file is available under two different licenses:
+* - Pimcore Open Core License (POCL)
+* - Pimcore Commercial License (PCL)
+* Full copyright and license information is available in
+* LICENSE.md which is distributed with this source code.
+*
+*  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+*  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
+*/
+
+import { App, type ModalFuncProps } from 'antd'
+import { Icon } from '@Pimcore/components/icon/icon'
+import React, { useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+
+type ConfigUpdate = ModalFuncProps | ((prevConfig: ModalFuncProps) => ModalFuncProps)
+
+interface ContentAware {
+  content: string
+}
+
+export interface UseSimpleModalResponse {
+  info: (props: ContentAware) => { destroy: () => void, update: (configUpdate: ConfigUpdate) => void }
+  error: (props: ContentAware) => { destroy: () => void, update: (configUpdate: ConfigUpdate) => void }
+  warn: (props: ContentAware) => { destroy: () => void, update: (configUpdate: ConfigUpdate) => void }
+}
+
+export const useSimpleModal = (): UseSimpleModalResponse => {
```

Comment:
Okay, I misunderstood that. I thought this would be the way. Nevertheless, I would go with this solution since it's cleaner.
I suggest reworking the original model to the new (and much smoother) approach.

Reply:
Yes, if you find the time feel free to rework the useModal() hook to the new approach 👍